dev/com.ibm.ws.security.oidc.server_fat/fat/src/com/ibm/ws/security/openidconnect/server/fat/BasicTests/OIDC/OIDCJWKRotationTest.java

```diff
@@ -0,0 +1,115 @@
+
+        // Initial request to get keys
+        WebResponse firstResponse = genericInvokeEndpoint(_testName, wc, null, updatedTestSettings.getJwkEndpt(),
+                Constants.GETMETHOD, Constants.INVOKE_JWK_ENDPOINT, null, null, expectations);
+
+        List<String> firstKeys = Json.createReader(new StringReader(AutomationTools.getResponseText(firstResponse)))
+                .readObject().getJsonArray("keys").stream().map(JsonValue::toString).collect(Collectors.toList());
+
+        // Repeat request, should be the same key
+        WebResponse secondResponse = genericInvokeEndpoint(_testName, wc, null, updatedTestSettings.getJwkEndpt(),
+                Constants.GETMETHOD, Constants.INVOKE_JWK_ENDPOINT, null, null, expectations);
+
+        List<String> secondKeys = Json.createReader(new StringReader(AutomationTools.getResponseText(secondResponse)))
+                .readObject().getJsonArray("keys").stream().map(JsonValue::toString).collect(Collectors.toList());
+
+        Assert.assertEquals(firstKeys, secondKeys);
+    }
+
+    // @Test
+    public void testJWKRotation_1m_ValidRotationTime_Success() throws Exception {
+        Log.entering(thisClass, _testName);
+        // Create the conversation object which will maintain state for us
+        WebConversation wc = new WebConversation();
+        TestSettings updatedTestSettings = testSettings.copyTestSettings();
+
+        updatedTestSettings.setJwkEndpt(updatedTestSettings.getJwkEndpt().replaceAll("OidcConfigSample", "OidcConfigSample_jwkRotationTime_1s"));
+
+        List<validationData> expectations = vData.addSuccessStatusCodes(null);
+        expectations = vData.addResponseExpectation(expectations, Constants.INVOKE_JWK_ENDPOINT, "Did not find JWKs in the response.", "\"keys\":");
+
+        // Initial request to get keys
+        WebResponse firstResponse = genericInvokeEndpoint(_testName, wc, null, updatedTestSettings.getJwkEndpt(),
+                Constants.GETMETHOD, Constants.INVOKE_JWK_ENDPOINT, null, null, expectations);
+
+        List<String> firstKeys = Json.createReader(new StringReader(AutomationTools.getResponseText(firstResponse)))
+                .readObject().getJsonArray("keys").stream().map(JsonValue::toString).collect(Collectors.toList());
+
+        // Wait rotation time
+        sleep(60000);
+
+        // Keys should not match
+        WebResponse secondResponse = genericInvokeEndpoint(_testName, wc, null, updatedTestSettings.getJwkEndpt(),
+                Constants.GETMETHOD, Constants.INVOKE_JWK_ENDPOINT, null, null, expectations);
+
+        List<String> secondKeys = Json.createReader(new StringReader(AutomationTools.getResponseText(secondResponse)))
+                .readObject().getJsonArray("keys").stream().map(JsonValue::toString).collect(Collectors.toList());
+
+        Assert.assertTrue(firstKeys.size() == secondKeys.size()
+            && !firstKeys.containsAll(secondKeys) && !secondKeys.containsAll((firstKeys)));
```

Comment:
I'd recommend splitting these out into individual assertions so we know exactly which part failed.
```suggestion
        Assert.assertEquals("Did not get the same number of keys in the second JWK request as the first request. firstKeys.size(), secondKeys.size());
        Assert.assertFalse("The first set of JWKs should not have contained the same keys as those from the second JWK request.", firstKeys.containsAll(secondKeys));
        Assert.assertFalse("The second set of JWKs should not have contained the same keys as those from the first JWK request.", secondKeys.containsAll(firstKeys));
```